src/add/operator/buffer.ts

```diff
@@ -4,4 +4,8 @@ import {buffer} from '../../operator/buffer';
 
 Observable.prototype.buffer = buffer;
 
-export var _void: void;
\ No newline at end of file
+declare module '../../Observable' {
+  interface Observable<T> {
+    buffer: (closingNotifier: Observable<any>) => Observable<T[]>;
```

Comment:
@masaeedu you're right, because of `buffer` expects there to be a typed this context, that is `T`.  So we'd have to add `BufferSiganture<T>` once `BufferSiganture<T>` is added.  Definitely not blocking.